cmd/cluster_kubeconfig.go

```diff
@@ -45,41 +53,153 @@ Example 4: hetzner-kube cluster kubeconfig -n my-cluster -p > my-conf.yaml # pri
 
 		FatalOnError(err)
 
-		printContent, _ := cmd.Flags().GetBool("print")
-		force, _ := cmd.Flags().GetBool("force")
-
-		if printContent {
-			fmt.Println(kubeConfigContent)
+		// get sanitized kubeconfig
+		// we need isSanitized flag to ensure we do not want do a merge if we fail to sanitize config
+		// if it fails we simply print out config as it is
+		isSanitized := false
+		newKubeConfig, err := sanitizeKubeConfig(kubeConfigContent, provider.GetCluster().Name, "hetzner")
+		if err != nil {
+			log.Printf("KubeConfig sanitise process failed, default config will be used instead. Error: %s", err.Error())
 		} else {
-			fmt.Println("create file")
+			kubeConfigContent = newKubeConfig
+			isSanitized = true
+		}
 
-			usr, _ := user.Current()
-			dir := usr.HomeDir
-			path := fmt.Sprintf("%s/.kube", dir)
+		if merge, _ := cmd.Flags().GetBool("merge"); merge && isSanitized {
 
-			if _, err := os.Stat(path); os.IsNotExist(err) {
-				os.MkdirAll(path, 0755)
+			if err = mergeKubeConfig(kubeConfigContent); err != nil {
+				log.Fatalf("During merge we encountered the problem: %s", err.Error())
 			}
+			log.Printf("KubeConfig successfully merged!")
+			return
+		}
 
-			// check if there already is an existing config
-			kubeconfigPath := fmt.Sprintf("%s/config", path)
-			if _, err := os.Stat(kubeconfigPath); !force && err == nil {
-				fmt.Println("There already exists a kubeconfig. Overwrite? (use -f to suppress this question) [yN]:")
-				r := bufio.NewReader(os.Stdin)
-				answer, err := r.ReadString('\n')
-				FatalOnError(err)
-				if !strings.ContainsAny(answer, "yY") {
-					log.Fatalln("aborted")
-				}
-			}
+		if save, _ := cmd.Flags().GetBool("save"); save {
```

Comment:
Ignored error

Reply:
same here, if they are defined, no error should occur